File: jersey/src/main/java/com/example/ClassUnit.java
package com.example;

import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
public class ClassUnit {
  public Integer id;
  public String name;
  public String enlistedOn;
  public String leftOn;

  public ClassUnit() {}

  public ClassUnit(
    String id,
    String name,
    String enlistedOn,
    String leftOn
  ) {
    this.id = id == null ? null : new Integer(id);
    this.name = name;
    this.enlistedOn = enlistedOn;
    this.leftOn = leftOn;
  }
}

File: kami/class_unit.go
package main

import "time"

func NewClassUnit() *ClassUnit {
	return &ClassUnit{}
}

type ClassUnit struct {
	Id         *int32     `json:"id,omitempty"`
	Name       *string    `json:"name,omitempty"`
	EnlistedOn *time.Time `json:"enlistedOn,omitempty"`
	LeftOn     *time.Time `json:"leftOn,omitempty"`
}
